Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -194,6 +194,135 @@ class BoolSetting: Setting {
     }
 }
 
+// A helper class for a setting backed by a UITextField.
+// This takes an optional settingIsValid and settingDidChange callback
+// If settingIsValid returns false, the Setting will not change and the text remains red.
+class StringSetting: Setting, UITextFieldDelegate {
+
+    let prefKey: String
+
+    private let prefs: Prefs
+    private let defaultValue: String?
+    private let placeholder: String?
+    private let settingDidChange: ((String?) -> Void)?
+    private let settingIsValid: ((String?) -> Bool)?
+
+    let textField = UITextField()
+
+    init(prefs: Prefs, prefKey: String, defaultValue: String? = nil, placeholder: String? = nil, accessibilityIdentifier: String? = nil, settingIsValid isValueValid: ((String?) -> Bool)? = nil, settingDidChange: ((String?) -> Void)? = nil) {
```

Comment:
I'd make `accessibilityIdentifier` non-optional unless you can think of a case where we wouldn't want the field to be accessible.

Same for `placeholder`, perhaps?